packages/twenty-server/src/modules/serverless/workflow-actions/code.workflow-action.ts

```diff
@@ -1,28 +1,24 @@
 import { Injectable } from '@nestjs/common';
 
+import { WorkflowAction } from 'src/modules/workflow/workflow-executor/interfaces/workflow-action.interface';
+
 import { ServerlessFunctionService } from 'src/engine/metadata-modules/serverless-function/serverless-function.service';
 import { ScopedWorkspaceContextFactory } from 'src/engine/twenty-orm/factories/scoped-workspace-context.factory';
-import { WorkflowActionResult } from 'src/modules/workflow/workflow-executor/types/workflow-action-result.type';
-import { WorkflowCodeStep } from 'src/modules/workflow/workflow-executor/types/workflow-action.type';
 import {
   WorkflowStepExecutorException,
   WorkflowStepExecutorExceptionCode,
 } from 'src/modules/workflow/workflow-executor/exceptions/workflow-step-executor.exception';
+import { WorkflowActionResult } from 'src/modules/workflow/workflow-executor/types/workflow-action-result.type';
+import { WorkflowCodeStepInput } from 'src/modules/workflow/workflow-executor/types/workflow-step-settings.type';
 
 @Injectable()
-export class CodeWorkflowAction {
+export class CodeWorkflowAction implements WorkflowAction {
   constructor(
     private readonly serverlessFunctionService: ServerlessFunctionService,
     private readonly scopedWorkspaceContextFactory: ScopedWorkspaceContextFactory,
   ) {}
 
-  async execute({
-    step,
-    payload,
-  }: {
-    step: WorkflowCodeStep;
-    payload?: object;
-  }): Promise<WorkflowActionResult> {
+  async execute(payload: WorkflowCodeStepInput): Promise<WorkflowActionResult> {
```

Comment:
`payload` might not be the proper naming finally. What about `workflowStepInput` or `actionInput` or just `input`?

Reply:
I will go with actionInput there!